Shortcodes: Merges the polldaddy shortcode changes from WP.com

zinigor opened the pull request:
This is a fairly large merge that brings the wpcom shortcode version in sync with the Jetpack version. Should fix #2079.

jeherve commented on modules/shortcodes/polldaddy.php:
```diff
@@ -105,16 +185,20 @@ function polldaddy_shortcode( $atts ) {
 PDRTJS_settings_{$rating}{$item_id}={$settings};
 //--><!]]></script>
-<script type="text/javascript" charset="UTF-8" src="http://i.polldaddy.com/ratings/rating.js"></script>
+<script type="text/javascript" charset="UTF-8" src="{$rating_js_file}"></script>
```
Shouldn't this be escaped with `esc_js()`?

zinigor replied:
Couldn't hurt, but in this case we should be using `esc_url()`, shouldn't we? Anyway, thanks, will fix!

jeherve commented on modules/shortcodes/polldaddy.php:
```diff
@@ -214,21 +313,26 @@ function polldaddy_shortcode( $atts ) {
 
 				if ( empty( $title ) ) {
-					$title = 'Take Our Survey';
+					$title = __( 'Take Our Survey' );
```
There is a missing Jetpack textdomain here.

zinigor replied:
Good catch, thank you!

jeherve commented on modules/shortcodes/polldaddy.php:
```diff
@@ -297,31 +402,34 @@ function polldaddy_shortcode( $atts ) {
+			if ( is_ssl() )
+				$rating_js_file = "https://polldaddy.com/js/rating/rating.js";
+			else
+				$rating_js_file = "http://i0.poll.fm/js/rating/rating.js";
```
Do we need to escape the URL used in js here as well?

dereksmart replied:
I don't believe that is necessary here

jeherve commented on modules/shortcodes/polldaddy.php:
```diff
@@ -13,12 +14,91 @@ class PolldaddyShortcode {
+  if ( !d.getElementById( j ) ) {
+    var pd = d.createElement( c ), s;
+    pd.id = j;
+    pd.src = ( 'https:' == d.location.protocol ) ? '{$embed_src_ssl}' : '{$embed_src}';
```
Should the URLs be escaped with `esc_js` here as well?

zinigor replied:
Yes, just the same, we may use `esc_url` just in case.

kraftbj commented on modules/shortcodes/polldaddy.php:
```diff
@@ -13,12 +14,91 @@ class PolldaddyShortcode {
+				
+				if ( $id > 0 ) {
+					$content = str_replace( $match[0], "[polldaddy poll=$id]", $content );
+					do_action( 'jetpack_embed_to_shortcode', 'polldaddy', $url );
```
Need to add preceding
`/** This action is documented in modules/shortcodes/youtube.php */`

zinigor commented:
@kraftbj Thanks for the doc hint, added.
@jeherve Thanks for spotting the missing textdomain, fixed. Also, I have been thinking about the escaping, and I don't see how it is useful here. We control the strings and we can see that they do not need any escaping. I wanted to add it just in case, but with HEREDOC notation it will be rather awkward, so let's just leave it like that.

dereksmart commented on modules/shortcodes/polldaddy.php:
```diff
@@ -13,12 +14,92 @@ class PolldaddyShortcode {
+		// Compress it a bit
+		$include = $this->compress_it( $include );
+
+		$placeholder = '<div class="pd-embed" data-settings="'.esc_attr( json_encode( $settings ) ).'"></div>';
```
Should be whitespace around the string operators

dereksmart commented on modules/shortcodes/polldaddy.php:
```diff
@@ -13,12 +14,92 @@ class PolldaddyShortcode {
+		$include = $this->compress_it( $include );
+
+		$placeholder = '<div class="pd-embed" data-settings="'.esc_attr( json_encode( $settings ) ).'"></div>';
+		if ( $type === 'button' )
```
Yoda conditionals plz

dereksmart replied:
Brackets for conditionals please

dereksmart commented on modules/shortcodes/polldaddy.php:
```diff
@@ -13,12 +14,92 @@ class PolldaddyShortcode {
+
+		$js_include = $placeholder."\n";
+		$js_include .= '<script type="text/javascript"><!--//--><![CDATA[//><!--'."\n";
+		$js_include .= $include."\n";
```
Should be whitespace around the string operators

dereksmart commented on modules/shortcodes/polldaddy.php:
```diff
@@ -13,12 +14,92 @@ class PolldaddyShortcode {
+		if ( $type === 'button' )
+			$placeholder = '<a class="pd-embed" href="'.esc_attr( $survey_link ).'" data-settings="'.esc_attr( json_encode( $settings ) ).'">'.esc_html( $settings['title'] ).'</a>';
+
+		$js_include = $placeholder."\n";
```
Should be whitespace around the string operators

dereksmart commented on modules/shortcodes/polldaddy.php:
```diff
@@ -13,12 +14,92 @@ class PolldaddyShortcode {
+		$js_include .= "//--><!]]></script>\n";
+
+		if ( $type !== 'button' )
+			$js_include .= '<noscript>'.$survey_link."</noscript>\n";
```
Should be whitespace around the string operators

dereksmart commented on modules/shortcodes/polldaddy.php:
```diff
@@ -13,12 +14,92 @@ class PolldaddyShortcode {
+		$js_include .= $include."\n";
+		$js_include .= "//--><!]]></script>\n";
+
+		if ( $type !== 'button' )
```
Yoda conditionals & braces

dereksmart commented on modules/shortcodes/polldaddy.php:
```diff
@@ -13,12 +14,92 @@ class PolldaddyShortcode {
+	 */
+	function polldaddy_embed_to_shortcode( $content ) {
+				
+		if ( false === strpos( $content, 'polldaddy.com/p/' ) )
```
missing braces

dereksmart commented on modules/shortcodes/polldaddy.php:
```diff
@@ -13,12 +14,92 @@ class PolldaddyShortcode {
+		$regexes[] = '#&lt;script(?:[^&]|&(?!gt;))+?src="https?://(secure|static)\.polldaddy\.com/p/([0-9]+)\.js"(?:[^&]|&(?!gt;))*+&gt;\s*?&lt;/script&gt;\r?\n?(&lt;noscript&gt;.*?&lt;/noscript&gt;)?#i';
+	
+		foreach ( $regexes as $regex ) {
+			if ( ! preg_match_all( $regex, $content, $matches, PREG_SET_ORDER ) )
```
Braces

dereksmart commented on modules/shortcodes/polldaddy.php:
```diff
@@ -13,12 +14,92 @@ class PolldaddyShortcode {
+		 		continue;
+	
+			foreach ( $matches as $match ) {
+				if (!isset( $match[2] ) )
```
missing whitespace

dereksmart commented on modules/shortcodes/polldaddy.php:
```diff
@@ -105,16 +186,20 @@ function polldaddy_shortcode( $atts ) {
 
 			$item_id = esc_js( $item_id );
 
+			if ( is_ssl() )
```
Braces please

dereksmart commented on modules/shortcodes/polldaddy.php:
```diff
@@ -181,26 +258,49 @@ function polldaddy_shortcode( $atts ) {
 					}
 
+					// Force the normal style embed on single posts/pages otherwise it's not rendered on infinite scroll themed blogs ('infinite_scroll_render' isn't fired)
+					if ( is_singular() )
```
Braces

dereksmart commented on modules/shortcodes/polldaddy.php:
```diff
@@ -181,26 +258,49 @@ function polldaddy_shortcode( $atts ) {
 						$data = array( 'url' => $poll_js );
 
-						self::$scripts['poll'][] = $data;
+						self::$scripts['poll'][intval( $poll )] = $data;
```
missing spaces in the $poll bracket.  Also should we first define this `$poll = intval( $poll )`?

dereksmart commented on modules/shortcodes/polldaddy.php:
```diff
@@ -297,31 +403,34 @@ function polldaddy_shortcode( $atts ) {
 		$script = '';
 
 		if ( is_array( self::$scripts ) ) {
+			if ( is_ssl() )
```
Braces

dereksmart commented on modules/shortcodes/polldaddy.php:
```diff
@@ -297,31 +403,34 @@ function polldaddy_shortcode( $atts ) {
-<noscript>{$survey_link}</noscript>
-CONTAINER;
+
+				if ( empty( $settings ) )
```
Braces

dereksmart commented on modules/shortcodes/polldaddy.php:
```diff
@@ -214,21 +314,26 @@ function polldaddy_shortcode( $atts ) {
 						$title = $link_text;
 				}
 
+				if ( $type == 'banner' || $type == 'slider' )
```
Braces

dereksmart commented:
I can say I have tested it, and the shortcode works properly.  The code has a whole lot of code style problems, and lacks documentation.  That said I don't see any blatant security holes or anything.  I agree with @zinigor about escaping those URLs.  Those are only set by us in the code, so no real need there.

zinigor commented:
@dereksmart thanks for the review! I have fixed the issues you pointed out, the file should be fully style compliant now.

dereksmart commented:
Looks great! tested with some of my own polldaddy shortcodes and works well.

dereksmart commented on modules/shortcodes/polldaddy.php:
```diff
@@ -13,12 +14,107 @@ class PolldaddyShortcode {
+				if ( $id > 0 ) {
+					$content = str_replace( $match[0], "[polldaddy poll=$id]", $content );
+					/** This action is documented in modules/shortcodes/youtube.php */
+					do_action( 'jetpack_embed_to_shortcode', 'polldaddy', $url );
```
$url throws an `undefined` notice when using the embed script

zinigor commented:
@dereksmart Thanks for taking a look, I have fixed the bug you mentioned, should be good to go now.
